### DISCUSSION TITLE
Trackball BTU

### DISCUSSION THREAD

**Discussion by simonkaiser9** [joshreve/dactyl-keyboard#52] [Ideas]
I am currently working on the integration of these bearings very similarly to the Ploopy option:
https://www.boschrexroth.com/de/de/produkte/produktgruppen/lineartechnik/kugelrollen-und-toleranzringe/kugelrollen/r0530-mit-stahlblechgehause/r0530-1-mit-stahlblechgehause

https://github.com/ploopyco/classic-trackball/tree/master/hardware/Mechanicals-BTU-Mod

I just want to make sure this is not already a work in progress somewhere and wanted to know whether you‘d be interested in the addition to be merged back here when it is ready.

As you seem to be hard at work on the trackball integration I want to make sure to base my work on the best possible branch to start from.
Would you recommend anything other than the master at this point? Or should I wait a few more days(?) until your work is ready?

Best regards
Simon

**Comment by joshreve** [comment]
It is up to you.  I keep getting pulled into fixes so I don't know when I will have the trackball fully done right now.  The one thing you can definitely do is block out the space you will need.  I briefly looked into those since they are quite handy, but the size makes it challenging to fit.  If you start with a blocky layout to see how it fits, that would definitely be worthwhile.  I think you will have to figure out the optimal orientation so the BTUs don't interfere with the keys.  

I am trying to get the socket out into generated geometry instead of imported, but that will likely still be a couple weeks.  Once you block it out, you can look at where I am with the updated geometry.  Also, I would definitely want to roll it in if you get get them working.  I think the BTUs are quite a nice trackball mechanism.

**Reply by simonkaiser9** [reply]
OK, my first drafts already generated the socket using the parameters present in the config, so that should be fine and go into the intended direction.
I need to find more info on the sensor cutout and holder, to make those fit without reinventing the wheel…
